app/ante/eth.go

```diff
@@ -324,20 +309,16 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		// calculate the fees paid to validators based on gas limit and price
 		feeAmt := msgEthTx.Fee() // fee = gas limit * gas price
 
-		evmDenom := egcd.evmKeeper.GetParams(infCtx).EvmDenom
+		evmDenom := egcd.evmKeeper.GetParams(ctx).EvmDenom
 		fees := sdk.Coins{sdk.NewCoin(evmDenom, sdk.NewIntFromBigInt(feeAmt))}
 
 		// deduct the full gas cost from the user balance
-		if err := authante.DeductFees(egcd.bankKeeper, infCtx, signerAcc, fees); err != nil {
+		if err := authante.DeductFees(egcd.bankKeeper, ctx, signerAcc, fees); err != nil {
 			return ctx, stacktrace.Propagate(
 				err,
 				"failed to deduct full gas cost %s from the user %s balance", fees, msgEthTx.From,
 			)
 		}
-
-		// consume intrinsic gas for the current transaction. After runTx is executed on Baseapp, the
-		// application will consume gas from the block gas pool.
-		ctx.GasMeter().ConsumeGas(intrinsicGas, "intrinsic gas")
```

Comment:
Removing this block will cause us to not consume any gas if the transaction fails, but we'd still deduct the fees from the user. Wondering if this the behavior we want and if so, we should rename this decorator to `EthDeductFeesDecorator` or something like that and explain this logic on the comments

Reply:
Actually if the transaction fails, we are still consuming gas since we are never calling `resetAndConsumeGas` .

The only thing is that the gas consumed will be equals to cosmos gas ops instead of intrinsic gas + evm gas.

Reply:
yeah we should document this logic